packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx

```diff
@@ -0,0 +1,230 @@
+      GetTwoUserTopicRecommendations(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+        comment {
+          _id
+          contents {
+            html
+            plaintextMainText
+          }
+        }
+        recommendationReason
+        yourVote
+        theirVote
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit:4 },
+  });
+
+  if (!currentUser) return <></>;
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
+  const topicRecommendations = preTopicRecommendations?.filter(topic => topic.theirVote !== null);
+  const numRecommendations = topicRecommendations?.length || 0;
+  const numShown = isExpanded ? numRecommendations : 1
+  const numHidden = Math.max(0, numRecommendations - numShown);
+
+  return (
+    <div>
+      <div key={targetUser._id} className={ isExpanded ? classNames(classes.dialogueUserRow, classes.dialogueUserRowExpandedMobile) : classes.dialogueUserRow}>
+        <div className={ (isExpanded || numRecommendations === 0) ? classNames(classes.dialogueLeftContainer, classes.dialogueLeftContainerExpandedMobile, classes.dialogueLeftContainerNoTopics) : classes.dialogueLeftContainer}>
+          <div className={ classes.dialogueMatchCheckbox }>
+            <DialogueCheckBox
+              targetUserId={targetUser._id}
+              targetUserDisplayName={targetUser.displayName}
+              checkId={checkId}
+              isChecked={userIsChecked}
+              isMatched={userIsMatched}
+            />
+          </div>
+          <PostsItem2MetaInfo className={classes.dialogueMatchUsername}>
+            <UsersName
+              documentId={targetUser._id}
+              simple={false}
+            />
+          </PostsItem2MetaInfo>
+        </div>
+        {showSuggestedTopics && <div className={classes.topicRecommendationsList}>
+          {topicRecommendations?.slice(0,numShown).map((topic, index) => (
+            <div key={index} className={classes.suggestionRow}>
+              <p key={index} className={ isExpanded ? classes.debateTopicExpanded : classes.debateTopicCollapsed}>
+                {topic.theirVote === 'agree' && [<ReactionIcon key={index} size={13} react={"agree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>agrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] } 
+                {topic.theirVote === 'disagree' && [<ReactionIcon key={index} size={13} react={"disagree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>disagrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] }
+              </p>
```

Comment:
This is not how we do conditional classNames; see any usage of the `classNames` function in e.g. DialogueMatchingPage.  (You do
```
classNames(classes.unconditionalClassName, {
  [classes.conditionalClassName]: booleanVariableWhichIsTrueIfClassShouldBeAppliedAndFalseOtherwise
})
```

Reply:
(also this entire section is pretty hard to read; would be good to extract various bits to top-level variable assignments instead of doing it all inline)